source/core.py

```diff
@@ -419,6 +419,20 @@ def _handleNVDAModuleCleanupBeforeGUIExit():
 	brailleViewer.destroyBrailleViewer()
 
 
+def _initializeObjectCaches():

```

Comment:
Please add a docstring, explain:
- Why `foregroundWindow` is used
- Why `desktopObject` is still set, security concerns, and what are the future plans.